src/int/neg.rs

```diff
@@ -0,0 +1,113 @@
+//! [`Int`] negation-related operations.
+
+use subtle::CtOption;
+
+use crate::{ConstChoice, Int, Word};
+
+impl<const LIMBS: usize> Int<LIMBS> {
+    /// Perform the two's complement "negate" operation on this [`Int`]:
+    /// map `self` to `(self ^ 1111...1111) + 0000...0001` and return the carry.
+    ///
+    /// Note: a zero carry indicates `self == Self::ZERO`.
```

Comment:
I don't think it's true. When self is negative and not MIN, the carry would be zero as well, right?

Reply:
When we treat a carry as implicit "`limbs[LIMBS]`" then we would have:
* for positive signed numbers carry is -1 (i.e. 1111...1111)
* for negative (not MIN) carry would be 0 (i.e. 0000...0000)
* for MIN carry is 1 (i.e. 0000....0001).
Not sure if my understanding is inlined with what the carry for negation means, it's more like a convention I guess the same as what carry for add means (as opposed to borrow for example).

Reply:
@tarcieri opinion would be appreciated :)